packages/contracts/contracts/shuffle/ShuffleManager.sol

```diff
@@ -67,28 +67,90 @@ contract ShuffleManager is IShuffleStateManager, Storage, Ownable {
         _deck30EncVerifier = deck30EncVerifier;
         _deck5EncVerifier = deck5EncVerifier;
         decryptVerifier = IDecryptVerifier(decryptVerifier_);
+
+        INIT_X1 = [
```

Comment:
Consider storing in `memory` instead of `storage` to save gas.

Reply:
I don't get it, how to store in ``` memory```

Reply:
The current implementation stores all the data in `storage`, taking about 20000 * 52 = 1.04 M when writing and 2100*52 = 109k every time when reading.

Instead, we can have the following function which stores the initial X1 in code and avoids the `storage` gas cost: 
```solidity
function getInitX1() internal returns(uint256[52] memory) {
    return [...];
}
```